yajsapi/mid-level-api/activity/activity.ts

```diff
@@ -104,26 +109,88 @@ export class Activity {
     return true;
   }
 
-  async getState(): Promise<ActivityStateStateEnum> {
-    const { data } = await this.stateApi.getActivityState(this.id);
-    // TODO: catch and check error
-    if (data?.state?.[0] && data?.state?.[0] !== this.state) {
-      this.state = data.state[0];
+  async getState(): Promise<ActivityStateEnum> {
+    try {
+      const { data } = await this.stateApi.getActivityState(this.id);
+      if (data?.state?.[0] && data?.state?.[0] !== this.state) {
+        this.state = data.state[0];
+      }
+      return this.state;
+    } catch (error) {
+      this.logger?.warn(`Cannot query activity state: ${error}`);
+      throw error;
     }
-    return this.state;
   }
 
   private async end(error?: Error) {
     await this.api
       .destroyActivity(this.id, this.requestTimeout, { timeout: (this.requestTimeout + 1) * 1000 })
       .catch((error) => this.logger?.warn(`Got API Exception when destroying activity ${this.id}: ${error}`));
-    if (this.stateFetchIntervalId) clearInterval(this.stateFetchIntervalId);
-    await this.getState();
     if (error) this.logger?.debug("Activity ended with an error: " + error);
     else this.logger?.debug("Activity ended");
   }
 
   private async handleError(error, cmdIndex, retryCount, maxRetries) {
-    // todo
+    if (this.isTimeoutError(error)) {
```

Comment:
plus, shouldn't this `handleError` logic be pulled somewhere outside of the activity module? it seems generic enough, doesn't it?

Reply:
It is very possible .. but I will wait with it until I find the same case in the code.